test/runnable/auto_ref.d

```diff
@@ -0,0 +1,55 @@
+struct S { }
+
+bool isLvalue(S s) {
+    return false;
+}
+
+bool isLvalue(ref S s) {
+    return true;
+}
+
+void oof(         S s) { }
+void ofo(     ref S s) { }
+void foo(auto ref S s) { }
+
+int bar(auto ref S s) { return 1; }
+//int bar(     ref S s) { return 1; }
+int bar(         S s) { return 2; }
```

Comment:
It shouldn't be possible to overload `auto ref` with either `ref` or value parameter; you either define by-value and by-reference overload, or `auto ref` alone.